src/components/TokenSelection/TokenRow.tsx

```diff
@@ -0,0 +1,90 @@
+import classNames from "classnames";
+import { TokenInfo } from "@uniswap/token-lists";
+
+export type TokenRowProps = {
+  /**
+   * TokenInfo object
+   */
+  token: TokenInfo;
+  /**
+   * Balance of current token
+   */
+  balance: string;
+  /**
+   * onClick event, either setSignerToken or setSenderToken
+   */
+  setToken: (val: string) => void;
+  /**
+   * Whether counterpart token is already selected; Still allows user to select that token
+   */
+  selected?: boolean;
+  /**
+   * Whether current token is already selected; Prevents token row click
+   */
+  disabled?: boolean;
+  /**
+   * Removes token from the active tokens list.
+   */
+  removeActiveToken: any;
+  /**
+   * Flag that determines if this is a default token that is automatically active.
+   */
+  defaultToken: boolean;
+};
+
+const TokenRow = ({
+  token,
+  balance,
+  setToken,
+  removeActiveToken,
+  selected,
+  disabled,
+  defaultToken,
+}: TokenRowProps) => {
+  return (
+    <div
```

Comment:
This should be button, not a div. Then same as above, add disabled prop to button

Reply:
I noticed that there's a <span> (removeActiveToken) inside the component that functions like a button. So we have a button in a button right now which is not recommended. You have to wrap the component in a container and add the two buttons on the same level.

Reply:
If this is too much work then I could also do it when I do the styled-components rewrite.

Reply:
Would you be able to do it, or guide me to how this should be implemented? We can perhaps do it in a separate issue as well too

Reply:
Alright, let's keep it like this for now.